semantic_model_generator/validate_model.py

```diff
@@ -39,44 +47,42 @@ def validate(yaml_str: str, snowflake_account: str) -> None:
     # Validate the context length doesn't exceed max we can support.
     validate_context_length(model, throw_error=True)
 
-    connector = SnowflakeConnector(
-        account_name=snowflake_account,
-        max_workers=1,
-    )
     model_in_column_format = context_to_column_format(model)
 
+    if conn is None:
+        connector = SnowflakeConnector(
+            account_name=snowflake_account,
+            max_workers=1,
+        )
+        conn = connector.open_connection(db_name="")
+
     for table in model_in_column_format.tables:
         logger.info(f"Checking logical table: {table.name}")
         # Each table can be a different database/schema.
-        # Create new connection for each one.
-        with connector.connect(
-            db_name=table.base_table.database, schema_name=table.base_table.schema
-        ) as conn:
-            try:
-                validate_all_cols(table)
-                sqls = generate_select(table, 1)
-                # Run the query.
-                # TODO: some expr maybe expensive if contains aggregations or window functions. Move to EXPLAIN?
-                for sql in sqls:
-                    _ = conn.cursor().execute(sql)
-            except Exception as e:
-                raise ValueError(f"Unable to validate your semantic model. Error = {e}")
-            logger.info(f"Validated logical table: {table.name}")
+        set_database(conn, table.base_table.database)
+        set_schema(conn, table.base_table.schema)
+        try:
+            validate_all_cols(table)
+            sqls = generate_select(table, 1)
+            # Run the query.
+            # TODO: some expr maybe expensive if contains aggregations or window functions. Move to EXPLAIN?
+            for sql in sqls:
+                _ = conn.cursor().execute(sql)
+        except Exception as e:
+            raise ValueError(f"Unable to validate your semantic model. Error = {e}")
+        logger.info(f"Validated logical table: {table.name}")
 
     for vq in model.verified_queries:
         logger.info(f"Checking verified queries for: {vq.question}")
-        with connector.connect(
```

Comment:
I think setting the db/schema in this connection was a no-op since this was outside of the `for table in ...` statement above, so I just removed it.